portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,17 +35,44 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> messages = new ArrayList<String>(Arrays.asList(
-      "Welcome to my page!", "Porfolio time!", "This is my portfolio"
-  ));
+  private ArrayList<String> comments = new ArrayList<String>();
+  private int numComments = 3;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJson(messages);
+    Query query = new Query("Input").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(numComments));
```

Comment:
Question: Why only show a certain number of comments?

Reply:
The walkthrough asked us to, I think to get used to http params